src/test/rgw/rgw_multi.py

```diff
@@ -0,0 +1,375 @@
+import json
+
+class Cluster(object):
+    """ interface to run commands against a distinct ceph cluster """
+    def __init__(self):
+        pass
+
+    def admin(self, cmd, check_retcode = True, stdin = None):
+        """ radosgw-admin command """
+        raise NotImplementedError
+
+class Gateway(object):
+    """ interface to control a single radosgw instance """
+    def __init__(self, host = None, port = None, cluster = None, zone = None, connection = None):
+        self.host = host
+        self.port = port
+        self.cluster = cluster
+        self.zone = zone
+        self.connection = connection
+
+    def start(self, args = None):
+        """ start the gateway """
```

Comment:
neat, i didn't know about abc. are there concrete advantages to using it in this case? it's hard for me to tell based on the docs (and lack of python background)